File: dynamic-programming/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector <int>> ps;
        for(int i=0;i<n;i++){
            vector<int> col(i+1,1);
            for(int j=1;j<i;j++){
                col[j]=ps[i-1][j-1]+ps[i-1][j];
            }
            ps.push_back(col);
        }
        return ps;
    }
};

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans=new ArrayList<>();
        for(int i=0;i<numRows;i++){
            List<Integer> level=new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0||j==i){
                    level.add(1);
                }else{
                    int val=ans.get(i-1).get(j-1)+ans.get(i-1).get(j);
                    level.add(val);
                }
            }
            ans.add(level);
        }
        return ans;
    }
}